src/irmin-pack/unix/stats.ml

```diff
@@ -171,13 +171,72 @@ module Index = struct
   let export m = Metrics.state m
 end
 
-type t = { pack_store : Pack_store.stat; index : Index.stat }
+module File_manager_stats = struct
+  type Metrics.origin += File_manager_stats
 
-let s = { pack_store = Pack_store.init (); index = Index.init () }
+  type t = {
+    mutable dict_flushes : int;
+    mutable suffix_flushes : int;
+    mutable index_flushes : int;
+    mutable auto_dict : int;
+    mutable auto_suffix : int;
+    mutable auto_index : int;
+    mutable flush : int;
+  }
+  [@@deriving irmin]
+
+  (* NOTE return a new instance each time, since fields are mutable *)
+  let init' () =
+    {
+      dict_flushes = 0;
+      suffix_flushes = 0;
+      index_flushes = 0;
+      auto_dict = 0;
+      auto_suffix = 0;
+      auto_index = 0;
+      flush = 0;
+    }
+
+  (* type [stat] is an abstract type in stats.mli; FIXME what is it for? *)
+  type stat = t Metrics.t
+
+  let init () : stat =
+    Metrics.v ~origin:File_manager_stats ~name:"file_manager_metric"
+      ~initial_state:(init' ()) t
+
+  (* [export] reveals the [t] contained in the [Metrics.t] container *)
+  let export : stat -> t = fun m -> Metrics.state m
+
+  (* support [reset_stats] function below *)
+  let clear' (t : t) =
+    t.dict_flushes <- 0;
+    t.suffix_flushes <- 0;
+    t.index_flushes <- 0;
+    ()
+
+  let clear (t : stat) = clear' (export t)
+end
+
+type t = {
+  pack_store : Pack_store.stat;
+  index : Index.stat;
+  file_manager : File_manager_stats.stat; 
+  (* FIXME why not just expose [File_manager_stats.t] here? What is reason for only
```

Comment:
so that we can get all stats with a single call to `get ()`?

Reply:
The question is about why we need an abstract type stat, when we could just expose `File_manager_stats.t` directly. (I understand that we want to have all stats as fields in a record type.)